frontend/src/lib-common/utils/usePeriodicRefresh.ts

```diff
@@ -0,0 +1,56 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import axios, { AxiosInstance } from 'axios'
+import { useState } from 'react'
+import { useInterval } from './useInterval'
+
+const TEN_SECONDS_MILLIS = 10 * 1000
+
+const calculateNextThreshold = (minutes: number) =>
+  Date.now() + minutes * 60 * 1000
+
+export interface PeriodicRefreshOptions {
+  thresholdInMinutes: number
+}
+
+// triggers after the given threshold in minutes
+// if there has also been axios activity after the threshold
+export function usePeriodicRefresh(
+  client: AxiosInstance,
+  callback: () => void,
+  { thresholdInMinutes }: PeriodicRefreshOptions = {
+    thresholdInMinutes: 10
+  }
+): () => void {
+  const [nextThreshold, setNextThreshold] = useState(
+    calculateNextThreshold(thresholdInMinutes)
+  )
+  const [lastResponse, setLastResponse] = useState(Date.now())
+
+  function executeOnExceededThreshold() {
+    if (Date.now() >= nextThreshold && lastResponse >= nextThreshold) {
+      setNextThreshold(calculateNextThreshold(thresholdInMinutes))
+      callback()
+    }
+  }
+
+  useInterval(executeOnExceededThreshold, TEN_SECONDS_MILLIS)
+
+  const id = client.interceptors.response.use(
+    (res) => {
+      setLastResponse(Date.now())
+      return res
+    },
+    (err: unknown) => {
+      if (axios.isAxiosError(err) && err.response) {
+        setLastResponse(Date.now())
+      }
+      return Promise.reject(err)
+    }
+  )
+  return () => {
+    client.interceptors.response.eject(id)
+  }
```

Comment:
You need to wrap this in a `useEffect` if you want the interceptor eject to happen automatically on unmount.